### PR TITLE
Cancel cell task when closing related cell doc

### PR DESCRIPTION
**Pull request by DonJayamanne**
For #5925

### CODE REVIEW

**Review by rchiodo — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/cellExecution.ts [hunk 1]
```diff
@@ -150,11 +150,11 @@ export class CellExecution {
                 // No point keeping it alive, just chewing resources.
                 if (e === this.cell.document) {
                     this.request?.dispose(); // NOSONAR
-                }
-                if (this.started && !this._completed) {
-                    this.completedDueToCancellation().catch((ex) =>
+                    if (this.started && !this._completed) {
```
Comment (IanMatthewHuff) [inline src/client/datascience/jupyter/kernels/cellExecution.ts:153]: Ahh, that was a bad one. Nice fix.

**Review by IanMatthewHuff — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

File: src/client/datascience/jupyter/kernels/cellExecution.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/jupyter/kernels/cellExecution.ts:153]: Yes, very bad.

**Comment by grudloff** [comment]
Hello! This should fix issue #5877?

**Comment by DonJayamanne** [comment]
Oh yes, sorry, I'll update the change log

**Pull request merged**